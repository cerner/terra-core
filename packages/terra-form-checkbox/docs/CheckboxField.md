# Checkbox Field

CheckboxField is a component of terra-form-checkbox that provides a container for rendering related checkboxes. CheckboxField include displays for help text, error text, required or optional indication, and invalid states.


## Usage

```jsx
import React from 'react';
import Checkbox from 'terra-form-checkbox';
import CheckboxField from 'terra-form-checkbox/lib/CheckboxField';

<CheckboxField
  isInline
  isInvalid
  error="This is not a valid choice"
  help="One of these must be filled out"
>
  <Checkbox id="first-checkbox" labelText="First Checkbox" name="example" defaultChecked />
  <Checkbox id="second-checkbox" labelText="Second Checkbox" name="example" />
</CheckboxField>
```

## Component Features

Uncomment supported features.
 * [Cross-Browser Support](https://github.com/cerner/terra-core/wiki/Component-Features#cross-browser-support)
 * [Responsive Support](https://github.com/cerner/terra-core/wiki/Component-Features#responsive-support)
 * [Mobile Support](https://github.com/cerner/terra-core/wiki/Component-Features#mobile-support)
 * [LTR/RTL Support](https://github.com/cerner/terra-core/wiki/Component-Features#ltr--rtl-support)
