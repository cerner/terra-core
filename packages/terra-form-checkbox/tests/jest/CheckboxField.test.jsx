import React from 'react';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { shallowWithIntl } from 'terra-enzyme-intl';
import CheckboxField from '../../src/CheckboxField';

window.matchMedia = () => ({ matches: true });

it('should render a default checkbox field', () => {
  const checkBox = (<CheckboxField legend="Default CheckboxField" />);
  const wrapper = shallowWithIntl(checkBox);
  expect(wrapper).toMatchSnapshot();
});

it('should render a default checkbox field even if it has an undefined child', () => {
  const undefinedChild = undefined;
  const checkBox = (<CheckboxField legend="Default CheckboxField">{ undefinedChild }</CheckboxField>);
  const wrapper = shallowWithIntl(checkBox);
  expect(wrapper).toMatchSnapshot();
});

it('should render an invalid checkbox field', () => {
  const checkBox = (<CheckboxField legend="Invalid CheckboxField" isInvalid error="Test Error" />);
  const wrapper = shallowWithIntl(checkBox);
  expect(wrapper).toMatchSnapshot();
});

it('should render a help message', () => {
  const checkBox = (<CheckboxField legend="Help CheckboxField" help="This will help up determine how many chairs to request" />);
  const wrapper = shallowWithIntl(checkBox);
  expect(wrapper).toMatchSnapshot();
});

it('should render an optional part on the label', () => {
  const checkBox = (<CheckboxField legend="Optional CheckboxField" showOptional />);
  const wrapper = shallowWithIntl(checkBox);
  expect(wrapper).toMatchSnapshot();
});

it('should render required checkbox field', () => {
  const checkBox = (<CheckboxField legend="Required CheckboxField" required />);
  const wrapper = shallowWithIntl(checkBox);
  expect(wrapper).toMatchSnapshot();
});

it('should hide the required indicator when requested', () => {
  const checkBox = (<CheckboxField legend="Hidden Required CheckboxField" required hideRequired />);
  const wrapper = shallowWithIntl(checkBox);
  expect(wrapper).toMatchSnapshot();
});

it('should display the required icon for fields with hideRequired, but have a state of invalid', () => {
  const checkBox = (<CheckboxField legend="Hidden Required CheckboxField" required hideRequired isInvalid />);
  const wrapper = shallowWithIntl(checkBox);
  expect(wrapper).toMatchSnapshot();
});

it('should render the legend with custom attributes properly', () => {
  const checkBox = (<CheckboxField legend="Custom Legend CheckboxField" legendAttrs={{ class: 'application-legend' }} />);
  const wrapper = shallowWithIntl(checkBox);
  expect(wrapper).toMatchSnapshot();
});

it('should hide the legend when requested', () => {
  const checkBox = (<CheckboxField legend="Hidden Legend legend" legendAttrs={{ class: 'application-legend' }} isLegendHidden />);
  const wrapper = shallowWithIntl(checkBox);
  expect(wrapper).toMatchSnapshot();
});
