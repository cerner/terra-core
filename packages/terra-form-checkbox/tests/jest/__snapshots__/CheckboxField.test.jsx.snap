// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display the required icon for fields with hideRequired, but have a state of invalid 1`] = `
<fieldset
  className="checkbox-field"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <legend
    className="legend-group"
    id="terra-checkbox-field-description-10"
  >
    <div
      className="legend"
    >
      <span
        className="error-icon"
      />
      <span
        className="required"
      >
        *
      </span>
      Hidden Required CheckboxField
    </div>
  </legend>
</fieldset>
`;

exports[`should hide the legend when requested 1`] = `
<fieldset
  className="checkbox-field"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <legend
    className="legend-group legend-group-hidden"
    id="terra-checkbox-field-description-12"
  >
    <div
      class="application-legend"
      className="legend"
    >
      Hidden Legend legend
      <span
        className="error-icon-hidden"
      />
    </div>
  </legend>
</fieldset>
`;

exports[`should hide the required indicator when requested 1`] = `
<fieldset
  className="checkbox-field"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <legend
    className="legend-group"
    id="terra-checkbox-field-description-9"
  >
    <div
      className="legend"
    >
      Hidden Required CheckboxField
      <span
        className="required-hidden"
      >
        *
      </span>
      <span
        className="error-icon-hidden"
      />
    </div>
  </legend>
</fieldset>
`;

exports[`should render a default checkbox field 1`] = `
<fieldset
  className="checkbox-field"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <legend
    className="legend-group"
    id="terra-checkbox-field-description-1"
  >
    <div
      className="legend"
    >
      Default CheckboxField
      <span
        className="error-icon-hidden"
      />
    </div>
  </legend>
</fieldset>
`;

exports[`should render a default checkbox field even if it has an undefined child 1`] = `
<fieldset
  className="checkbox-field"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <legend
    className="legend-group"
    id="terra-checkbox-field-description-2"
  >
    <div
      className="legend"
    >
      Default CheckboxField
      <span
        className="error-icon-hidden"
      />
    </div>
  </legend>
</fieldset>
`;

exports[`should render a help message 1`] = `
<fieldset
  className="checkbox-field"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <legend
    className="legend-group"
    id="terra-checkbox-field-description-5"
  >
    <div
      className="legend"
    >
      Help CheckboxField
      <span
        className="error-icon-hidden"
      />
    </div>
  </legend>
  <div
    className="help-text"
    id="terra-checkbox-field-description-help-6"
  >
    This will help up determine how many chairs to request
  </div>
</fieldset>
`;

exports[`should render an invalid checkbox field 1`] = `
<fieldset
  className="checkbox-field"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <legend
    className="legend-group"
    id="terra-checkbox-field-description-3"
  >
    <div
      className="legend"
    >
      <span
        className="error-icon"
      />
      Invalid CheckboxField
    </div>
  </legend>
  <div
    className="error-text"
    id="terra-checkbox-field-description-error-4"
  >
    Test Error
  </div>
</fieldset>
`;

exports[`should render an optional part on the label 1`] = `
<fieldset
  className="checkbox-field"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <legend
    className="legend-group"
    id="terra-checkbox-field-description-7"
  >
    <div
      className="legend"
    >
      Optional CheckboxField
      <span
        className="optional"
      >
        Terra.form.field.optional
      </span>
      <span
        className="error-icon-hidden"
      />
    </div>
  </legend>
</fieldset>
`;

exports[`should render required checkbox field 1`] = `
<fieldset
  className="checkbox-field"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <legend
    className="legend-group"
    id="terra-checkbox-field-description-8"
  >
    <div
      className="legend"
    >
      <span
        className="required"
      >
        *
      </span>
      Required CheckboxField
      <span
        className="error-icon-hidden"
      />
    </div>
  </legend>
</fieldset>
`;

exports[`should render the legend with custom attributes properly 1`] = `
<fieldset
  className="checkbox-field"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <legend
    className="legend-group"
    id="terra-checkbox-field-description-11"
  >
    <div
      class="application-legend"
      className="legend"
    >
      Custom Legend CheckboxField
      <span
        className="error-icon-hidden"
      />
    </div>
  </legend>
</fieldset>
`;
